core/services/gateway/handlers/functions/allowlist/allowlist.go

```diff
@@ -68,12 +81,29 @@ func NewOnchainAllowlist(client evmclient.Client, config OnchainAllowlistConfig,
 	if config.ContractVersion != 1 {
 		return nil, fmt.Errorf("unsupported contract version %d", config.ContractVersion)
 	}
+
+	if config.StoredAllowlistBatchSize == 0 {
+		lggr.Info("StoredAllowlistBatchSize not specified, using default size: ", defaultStoredAllowlistBatchSize)
+		config.StoredAllowlistBatchSize = defaultStoredAllowlistBatchSize
+	}
+
+	if config.OnchainAllowlistBatchSize == 0 {
+		lggr.Info("OnchainAllowlistBatchSize not specified, using default size: ", defaultOnchainAllowlistBatchSize)
+		config.OnchainAllowlistBatchSize = defaultOnchainAllowlistBatchSize
+	}
+
+	if config.FetchingDelayInRangeSec == 0 {
+		lggr.Info("FetchingDelayInRangeSec not specified, using default size: ", defaultOnchainAllowlistBatchSize)
+		config.FetchingDelayInRangeSec = defaultFetchingDelayInRangeSec
+	}
+
```

Comment:
As a sanity check, it might make sense to add a check that `FetchingDelayInRangeSec` is less than `UpdateFrequencySec`